byway/src/main/webapp/routepage.js

```diff
@@ -229,10 +240,16 @@ function getRecsOnload() {
   clearRecs();
   fetch('/api/recs')
   .then(response => response.json())
-  .then((recommendations) => {
-     recommendations.forEach((rec)=>{
-      recs.add(rec);
-    })
+  .then(async (recommendations) => {
+    for (let rec of recommendations){
```

Comment:
Same here, consider naming the rec variable to be id-specific

Reply:
Changed!